src/loki/worker.cc

```diff
@@ -245,6 +245,7 @@ loki_worker_t::loki_worker_t(const boost::property_tree::ptree& config,
 }
 
 void loki_worker_t::cleanup() {
+  service_worker_t::cleanup();
```

Comment:
same thing, the base class actually has stuff to clean up, in this case it ships all the queued metrics to the stats daemon if so configured